### DISCUSSION TITLE
Template for RFC/RFP

### DISCUSSION THREAD

**Discussion by mimir-d** [opencomputeproject/ocp-diag-core#631] [Ideas]
We should create a template for people to propose changes to the spec and manage the process.

**Comment by mimir-d** [comment]
@dhawkes I think you had some relevant ideas about what the fields for the template should be. Would be nice to continue here.
@dframe-google Please share your ideas from the call as well.

Thanks

**Reply by mimir-d** [reply]
Transcribing some of the discussion from last week's call.

Fields proposal:
- abstract: concise form of the proposal, for quick reading context
- motivation: _why?_ we need this change
- proposed change + examples: _what?_ is the actual change and relevant showcasing
- alternatives: [optional] if there's other ways of resolving the _why?_; drawbacks
- unresolved questions: [optional] self-explanatory

Regarding spec change process:
1. start a discussion item in ocp-diag-core Discussions tab
2. if necessary, assign a PR with proposed spec changes
3. gather consensus/feedback
4. core OCPTV team reviews the RFC, resulting in merge/no-merge/postpone

I would also propose the rust process as a reference here, we could follow the text for more details (or phrase our process in a similar fashion).
https://github.com/rust-lang/rfcs

**Reply by dframe-google** [reply]
Thanks for the rust link, I will check it out.  The general outline above looks good to me.
